items:
- uid: com.microsoft.azure.batch.protocol.models._batch_error_exception
  id: _batch_error_exception
  parent: com.microsoft.azure.batch.protocol.models
  children:
  - com.microsoft.azure.batch.protocol.models._batch_error_exception.BatchErrorException()
  - classcom.microsoft.azure.batch.protocol.models._batch_error_exception.BatchErrorException(final String)
  - classcom.microsoft.azure.batch.protocol.models._batch_error_exception.BatchErrorException(final String,final Throwable)
  - classcom.microsoft.azure.batch.protocol.models._batch_error_exception.BatchErrorException(final Throwable)
  - com.microsoft.azure.batch.protocol.models._batch_error_exception.getBody()
  - com.microsoft.azure.batch.protocol.models._batch_error_exception.getResponse()
  - com.microsoft.azure.batch.protocol.models._batch_error_exception.setBody(BatchError)
  - com.microsoft.azure.batch.protocol.models._batch_error_exception.setResponse(Response)
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  langs:
  - java
  name: BatchErrorException
  fullName: com.microsoft.azure.batch.protocol.models.BatchErrorException
  type: Class
  source:
    remote: &o0
      path: azure-batch/src/main/java/com/microsoft/azure/batch/protocol/models/BatchErrorException.java
      branch: master
      repo: https://github.com/Azure/azure-sdk-for-java.git
    path: azure-batch/src/main/java/com/microsoft/azure/batch/protocol/models/BatchErrorException.java
    startLine: 16
  package: com.microsoft.azure.batch.protocol.models
  summary: <p>Exception thrown for an invalid response with <xref href="com.microsoft.azure.batch.protocol.models._batch_error" data-throw-if-not-resolved="false"></xref> information. </p>
  syntax: &o1
    content: public class BatchErrorException extends RestException
  inheritance:
  - java.lang.Object
  - Exception
  - com.microsoft.rest._rest_exception
  inheritedMembers:
  - com.microsoft.rest._rest_exception.RestException()
  - com.microsoft.rest._rest_exception.RestException(String)
  - com.microsoft.rest._rest_exception.RestException(Throwable)
  - com.microsoft.rest._rest_exception.RestException(String,Throwable)
- uid: com.microsoft.azure.batch.protocol.models._batch_error_exception.BatchErrorException()
  id: BatchErrorException()
  parent: com.microsoft.azure.batch.protocol.models._batch_error_exception
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  langs:
  - java
  name: BatchErrorException()
  fullName: com.microsoft.azure.batch.protocol.models.BatchErrorException.BatchErrorException()
  type: Constructor
  source:
    remote: *o0
    path: azure-batch/src/main/java/com/microsoft/azure/batch/protocol/models/BatchErrorException.java
    startLine: 28
  package: com.microsoft.azure.batch.protocol.models
  summary: <p>Initializes a new instance of the <xref href="com.microsoft.azure.batch.protocol.models._batch_error_exception" data-throw-if-not-resolved="false"></xref> class. </p>
  syntax:
    content: public BatchErrorException()
- uid: classcom.microsoft.azure.batch.protocol.models._batch_error_exception.BatchErrorException(final String)
  id: BatchErrorException(final String)
  parent: com.microsoft.azure.batch.protocol.models._batch_error_exception
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  langs:
  - java
  name: BatchErrorException(final String message)
  fullName: com.microsoft.azure.batch.protocol.models.BatchErrorException.BatchErrorException(final String message)
  type: Constructor
  source:
    remote: *o0
    path: azure-batch/src/main/java/com/microsoft/azure/batch/protocol/models/BatchErrorException.java
    startLine: 34
  package: com.microsoft.azure.batch.protocol.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.batch.protocol.models._batch_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public BatchErrorException(final String message)
    parameters:
    - id: message
      type: 3d93591e
      description: <p>The exception message. </p>
- uid: classcom.microsoft.azure.batch.protocol.models._batch_error_exception.BatchErrorException(final String,final Throwable)
  id: BatchErrorException(final String,final Throwable)
  parent: com.microsoft.azure.batch.protocol.models._batch_error_exception
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  langs:
  - java
  name: BatchErrorException(final String message, final Throwable cause)
  fullName: com.microsoft.azure.batch.protocol.models.BatchErrorException.BatchErrorException(final String message, final Throwable cause)
  type: Constructor
  source:
    remote: *o0
    path: azure-batch/src/main/java/com/microsoft/azure/batch/protocol/models/BatchErrorException.java
    startLine: 43
  package: com.microsoft.azure.batch.protocol.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.batch.protocol.models._batch_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public BatchErrorException(final String message, final Throwable cause)
    parameters:
    - id: message
      type: 3d93591e
      description: <p>the exception message </p>
    - id: cause
      type: fa00b19f
      description: <p>exception that caused this exception to occur </p>
- uid: classcom.microsoft.azure.batch.protocol.models._batch_error_exception.BatchErrorException(final Throwable)
  id: BatchErrorException(final Throwable)
  parent: com.microsoft.azure.batch.protocol.models._batch_error_exception
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  langs:
  - java
  name: BatchErrorException(final Throwable cause)
  fullName: com.microsoft.azure.batch.protocol.models.BatchErrorException.BatchErrorException(final Throwable cause)
  type: Constructor
  source:
    remote: *o0
    path: azure-batch/src/main/java/com/microsoft/azure/batch/protocol/models/BatchErrorException.java
    startLine: 51
  package: com.microsoft.azure.batch.protocol.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.batch.protocol.models._batch_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public BatchErrorException(final Throwable cause)
    parameters:
    - id: cause
      type: fa00b19f
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.azure.batch.protocol.models._batch_error_exception.getBody()
  id: getBody()
  parent: com.microsoft.azure.batch.protocol.models._batch_error_exception
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  langs:
  - java
  name: getBody()
  fullName: BatchError com.microsoft.azure.batch.protocol.models.BatchErrorException.getBody()
  type: Method
  source:
    remote: *o0
    path: azure-batch/src/main/java/com/microsoft/azure/batch/protocol/models/BatchErrorException.java
    startLine: 67
  package: com.microsoft.azure.batch.protocol.models
  summary: >-
    <p>Gets the HTTP response body.</p>

    <p></p>
  syntax:
    content: public BatchError getBody()
    return:
      type: com.microsoft.azure.batch.protocol.models._batch_error
      description: <p>the response body </p>
- uid: com.microsoft.azure.batch.protocol.models._batch_error_exception.getResponse()
  id: getResponse()
  parent: com.microsoft.azure.batch.protocol.models._batch_error_exception
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  langs:
  - java
  name: getResponse()
  fullName: Response com.microsoft.azure.batch.protocol.models.BatchErrorException.getResponse()
  type: Method
  source:
    remote: *o0
    path: azure-batch/src/main/java/com/microsoft/azure/batch/protocol/models/BatchErrorException.java
    startLine: 59
  package: com.microsoft.azure.batch.protocol.models
  summary: >-
    <p>Gets information about the associated HTTP response.</p>

    <p></p>
  syntax:
    content: public Response getResponse()
    return:
      type: e9d34ed6
      description: <p>the HTTP response </p>
- uid: com.microsoft.azure.batch.protocol.models._batch_error_exception.setBody(BatchError)
  id: setBody(BatchError)
  parent: com.microsoft.azure.batch.protocol.models._batch_error_exception
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  langs:
  - java
  name: setBody(BatchError body)
  fullName: void com.microsoft.azure.batch.protocol.models.BatchErrorException.setBody(BatchError body)
  type: Method
  source:
    remote: *o0
    path: azure-batch/src/main/java/com/microsoft/azure/batch/protocol/models/BatchErrorException.java
    startLine: 83
  package: com.microsoft.azure.batch.protocol.models
  summary: >-
    <p>Sets the HTTP response body.</p>

    <p></p>
  syntax:
    content: public void setBody(BatchError body)
    parameters:
    - id: body
      type: com.microsoft.azure.batch.protocol.models._batch_error
      description: <p>the response body </p>
- uid: com.microsoft.azure.batch.protocol.models._batch_error_exception.setResponse(Response)
  id: setResponse(Response)
  parent: com.microsoft.azure.batch.protocol.models._batch_error_exception
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  langs:
  - java
  name: setResponse(Response response)
  fullName: void com.microsoft.azure.batch.protocol.models.BatchErrorException.setResponse(Response response)
  type: Method
  source:
    remote: *o0
    path: azure-batch/src/main/java/com/microsoft/azure/batch/protocol/models/BatchErrorException.java
    startLine: 75
  package: com.microsoft.azure.batch.protocol.models
  summary: >-
    <p>Sets the HTTP response.</p>

    <p></p>
  syntax:
    content: public void setResponse(Response response)
    parameters:
    - id: response
      type: e9d34ed6
      description: <p>the HTTP response </p>
references:
- uid: 3d93591e
  spec.java:
  - name: final String
    fullName: final String
- uid: fa00b19f
  spec.java:
  - name: final Throwable
    fullName: final Throwable
- uid: e9d34ed6
  spec.java:
  - name: Response
    fullName: Response
- uid: com.microsoft.rest._rest_exception.RestException()
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException()
  fullName: com.microsoft.rest.RestException.RestException()
  type: Constructor
  summary: <p>Initializes a new instance of the AutoRestException class. </p>
  syntax:
    content: public RestException()
- uid: com.microsoft.rest._rest_exception.RestException(String)
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException(String message)
  fullName: com.microsoft.rest.RestException.RestException(String message)
  type: Constructor
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(String message)
    parameters:
    - id: message
      type: "26831127"
      description: <p>The exception message. </p>
- uid: com.microsoft.rest._rest_exception.RestException(Throwable)
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException(Throwable cause)
  fullName: com.microsoft.rest.RestException.RestException(Throwable cause)
  type: Constructor
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(Throwable cause)
    parameters:
    - id: cause
      type: a04fafd7
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.rest._rest_exception.RestException(String,Throwable)
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException(String message, Throwable cause)
  fullName: com.microsoft.rest.RestException.RestException(String message, Throwable cause)
  type: Constructor
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(String message, Throwable cause)
    parameters:
    - id: message
      type: "26831127"
      description: <p>the exception message </p>
    - id: cause
      type: a04fafd7
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.rest._rest_exception
  parent: com.microsoft.rest
  href: com.microsoft.rest._rest_exception.yml
  name: RestException
  fullName: com.microsoft.rest.RestException
  type: Class
  summary: <p>Exception thrown for an invalid response with custom error information. </p>
  syntax:
    content: public class RestException
- uid: com.microsoft.azure.batch._create_tasks_terminated_exception
  parent: com.microsoft.azure.batch
  href: com.microsoft.azure.batch._create_tasks_terminated_exception.yml
  name: CreateTasksTerminatedException
  fullName: com.microsoft.azure.batch.CreateTasksTerminatedException
  type: Class
  summary: >-
    <p>

      <xref href="com.microsoft.azure.batch._create_tasks_terminated_exception" data-throw-if-not-resolved="false"></xref>. </p>
  syntax:
    content: public class CreateTasksTerminatedException
- uid: com.microsoft.azure.batch.protocol.models._batch_error
  parent: com.microsoft.azure.batch.protocol.models
  href: com.microsoft.azure.batch.protocol.models._batch_error.yml
  name: BatchError
  fullName: com.microsoft.azure.batch.protocol.models.BatchError
  type: Class
  summary: <p>An error response received from the <xref href="com.microsoft.azure._azure" data-throw-if-not-resolved="false"></xref> Batch service. </p>
  syntax:
    content: public class BatchError
- uid: com.microsoft.azure.batch.protocol.models._batch_error_exception
  parent: com.microsoft.azure.batch.protocol.models
  href: com.microsoft.azure.batch.protocol.models._batch_error_exception.yml
  name: BatchErrorException
  fullName: com.microsoft.azure.batch.protocol.models.BatchErrorException
  type: Class
  summary: <p>Exception thrown for an invalid response with <xref href="com.microsoft.azure.batch.protocol.models._batch_error" data-throw-if-not-resolved="false"></xref> information. </p>
  syntax: *o1
